pysb/simulator/scipyode.py

```diff
@@ -209,40 +200,52 @@ def __init__(self, model, tspan=None, initials=None, param_values=None,
             ydot = np.zeros(len(self.model.species))
 
             if self._compiler == 'cython':
+                self._compiler_directives = kwargs.pop(
+                    'cython_directives', self.default_cython_directives
+                )
+
                 if not Cython:
                     raise ImportError('Cython library is not installed')
-                self._code_eqs = code_eqs
-
-                def rhs(t, y, p):
-                    # note that the evaluated code sets ydot as a side effect
-                    Cython.inline(
-                        code_eqs, quiet=True,
-                        cython_compiler_directives=cython_directives)
 
-                    return ydot
+                rhs = _get_rhs(self._compiler,
+                               code_eqs,
+                               ydot=ydot,
+                               compiler_directives=self._compiler_directives
+                               )
 
                 with _set_cflags_no_warnings(self._logger):
                     rhs(0.0, self.initials[0], self.param_values[0])
             else:
                 # Weave
+                self._compiler_directives = []
+                # Inhibit weave C compiler warnings unless log
+                # level <= EXTENDED_DEBUG. Note that since the output goes
+                # straight to stderr rather than via the logging system, the
+                # threshold must be lower than DEBUG or else the Nose
+                # logcapture plugin will cause the warnings to be shown and
+                # tests will fail due to unexpected output.
+                if not self._logger.isEnabledFor(EXTENDED_DEBUG):
+                    self._compiler_directives.append('-w')
                 if not weave_inline:
                     raise ImportError('Weave library is not installed')
                 for arr_name in ('ydot', 'y', 'p'):
                     macro = arr_name.upper() + '1'
                     code_eqs = re.sub(r'\b%s\[(\d+)\]' % arr_name,
                                       '%s(\\1)' % macro, code_eqs)
 
-                def rhs(t, y, p):
-                    # note that the evaluated code sets ydot as a side effect
-                    weave_inline(code_eqs, ['ydot', 't', 'y', 'p'],
-                                 extra_compile_args=extra_compile_args)
-                    return ydot
+                    rhs = _get_rhs(self._compiler,
```

Comment:
This needs to be un-indented one level -- it's now inside the preceding loop!

Reply:
Oops, good catch!